connector/src/main/java/org/geysermc/connector/utils/LanguageUtils.java

```diff
@@ -0,0 +1,148 @@
+
+import org.geysermc.connector.GeyserConnector;
+
+import java.io.InputStream;
+import java.text.MessageFormat;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+public class LanguageUtils {
+
+    private static final Map<String, Properties> LOCALE_MAPPINGS = new HashMap<>();
+
+    private static String DEFAULT_LOCALE = (GeyserConnector.getInstance().getConfig().getDefaultLocale() != null ? GeyserConnector.getInstance().getConfig().getDefaultLocale() : "en_us");
+
+    static {
+        // Load it as a backup in case something goes really wrong
+        if (!"en_US".equals(cleanLocale(DEFAULT_LOCALE))) {
+            loadGeyserLocale("en_US");
+        }
+
+        // Load the default locale from config
+        loadGeyserLocale(DEFAULT_LOCALE);
+    }
+
+    /**
+     * Loads a Geyser locale from resources, if the file doesn't exist it just logs a warning
+     *
+     * @param locale Locale to load
+     */
+    public static void loadGeyserLocale(String locale) {
+        locale = cleanLocale(locale);
+
+        InputStream localeStream = LocaleUtils.class.getClassLoader().getResourceAsStream("languages/texts/" + locale + ".properties");
+
+        // Load the locale
+        if (localeStream != null) {
+            Properties localeProp = new Properties();
+            try {
+                localeProp.load(localeStream);
+            } catch (Exception e) {
+                throw new AssertionError(getLocaleStringLog("geyser.language.load_failed", locale), e);
+            }
+
+            // Insert the locale into the mappings
+            LOCALE_MAPPINGS.put(locale, localeProp);
+        } else {
+            if (locale.toLowerCase().equals(DEFAULT_LOCALE.toLowerCase())) {
+                // The default locale was invalid fallback to en_us
+                DEFAULT_LOCALE = "en_US";
+                loadGeyserLocale(DEFAULT_LOCALE);
+            } else {
+                GeyserConnector.getInstance().getLogger().warning(getLocaleStringLog("geyser.language.missing_file", locale));
+            }
+        }
+    }
+
+    /**
+     * Get a formatted language string with the default locale for Geyser
+     *
+     * @param key Language string to translate
+     * @param values Values to put into the string
+     * @return Translated string or the original message if it was not found in the given locale
+     */
+    public static String getLocaleStringLog(String key, Object... values) {
+        return getLocaleStringPly(key, DEFAULT_LOCALE, values);
+    }
+
+    /**
+     * Get a formatted language string with the given locale for Geyser
+     *
+     * @param key Language string to translate
+     * @param locale Locale to translate to
+     * @param values Values to put into the string
+     * @return Translated string or the original message if it was not found in the given locale
+     */
+    public static String getLocaleStringPly(String key, String locale, Object... values) {
```

Comment:
You mentioned that `Ply` is used in Garry's Mod, but for Minecraft it's probably better to change this to `getLocaleStringPlayer`.